File: solarbackend/src/main/java/solar/solarbackend/entity/ProjectPanel.java
package solar.solarbackend.entity;


import com.fasterxml.jackson.annotation.*;
import jakarta.persistence.*;
import lombok.Data;

@Entity

@Data
@Table(name = "Project_Panel")
public class ProjectPanel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "panel_id")
    private Long id;

    @Column(name = "direction" )
    private String direction;

    @Column(name = "serialnumber" )
    private String serialnumber;


    @ManyToOne
    @JoinColumn(name = "project_id" )
    private Project project;


    @ManyToOne
    @JoinColumn(name = "panel" )

    private Panels panel;
}

File: frontend/src/app/project-panel.ts
export class ProjectPanel {
    id?: number // Optional ID
    direction: string; // Use lowercase 'string'
    serialnumber: string;

    project: { id: number };
    panel: { id: number };

    constructor(
        direction: string = "",
        pid: number = 1,
        paID: number = 1,
        serialnumber: string = ""
    ) {
        this.direction = direction;
        this.serialnumber = serialnumber;
        this.project = { id: pid };
        this.panel = { id: paID };
    }
}
